lib/core/test/unit/Cardano/Wallet/Primitive/Types/UTxOIndexSpec.hs

```diff
@@ -531,6 +537,38 @@ prop_selectRandom_all_withAssetOnly u a = checkCoverage $ monadicIO $ do
     assert $ UTxOIndex.deleteMany (fst <$> selectedEntries) u == u'
     assert $ UTxOIndex.insertMany selectedEntries u' == u
 
+-- | Verify that priority order is respected when selecting with more than
+--   one filter.
+--
+prop_selectRandomWithPriority :: UTxOIndex -> Property
+prop_selectRandomWithPriority u =
+    forAll (genAssetIdSmallRange) $ \a1 ->
```

Comment:
Just wondering, are the two levels of forAll essential, or can one AssetId be a function parameter?

Reply:
Indeed, the first level could be replaced with a function parameter.

However, by using `forAll` for both, it's easy to see (at a glance) that both asset ids come from the same generator.